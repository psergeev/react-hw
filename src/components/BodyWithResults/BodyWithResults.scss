.results {
    background-color: #fdfffe;
    display: flex;
    flex-direction: column;

    .results-not-found {
        margin: auto 0 auto 0;
        text-align: center;
        font-size: 50px;
        font-weight: normal;
    }

    .results-list {
        padding: 30px 50px;
        display: flex;
    }
}