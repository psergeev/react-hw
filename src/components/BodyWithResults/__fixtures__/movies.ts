import { Movie } from '../../../App';

const movies: Movie[] = [{
    poster_path: 'https://image.tmdb.org/t/p/w500/97fNAi62HawGjWru7PvVmF7RAbU.jpg',
    title: 'Kill Bill: Vol. 1',
    tagline: 'Go for the kill.',
    release_date: '2003-10-10',
    vote_average: 7.8,
    runtime: 111,
    overview: 'An assassin is shot at the altar by her ruthless employer, ' +
        'Bill and other members of their assassination circle – but \'The Bride\' ' +
        'lives to plot her vengeance. Setting out for some payback, she makes a death ' +
        'list and hunts down those who wronged her, saving Bill for last.',
    genres: [
        'Action',
        'Crime',
    ],
}, {
    poster_path: 'https://image.tmdb.org/t/p/w500/au9lFA5a2ZnBKCzPbZQf00r7J64.jpg',
    title: 'Kill Bill: Vol. 2',
    tagline: 'The bride is back for the final cut.',
    release_date: '2004-04-16',
    vote_average: 7.7,
    runtime: 136,
    overview: 'The Bride unwaveringly continues on her roaring rampage of revenge against ' +
        'the band of assassins who had tried to kill her and her unborn child. She visits ' +
        'each of her former associates one-by-one, checking off the victims on her Death List ' +
        'Five until there\'s nothing left to do … but kill Bill.',
    genres: [
        'Action',
        'Crime',
        'Thriller',
    ],
}];

export default movies;
