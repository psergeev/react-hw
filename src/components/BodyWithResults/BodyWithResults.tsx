import * as React from 'react';
import { Movie } from '../../App';
import MovieCard from '../MovieCard';
import ResultsDescription from '../ResultsDescription';
import ResultsSorter from '../ResultsSorter';
import './BodyWithResults.scss';

interface Props {
    movies: Movie[];
    handleMovieCardClick: (title: string) => void;
    genre: string;
    isEmpty: boolean;
}

export default React.memo((props: Props) => {
    const movieCards = props.movies.map(movie => (
        <MovieCard
            handleMovieCardClick={props.handleMovieCardClick}
            movie={movie}
            key={movie.title}
        />
    ));

    const getResultList = () => {
        if (!props.movies.length && !props.isEmpty) {
            return <h2 className="results-not-found">No films found</h2>;
        }

        if (!props.isEmpty) {
            return (
                <div className="results-list">
                    {movieCards}
                </div>
            );
        }

        return '';
    };

    return (
        <section className="results">
            {props.genre ? <ResultsDescription genre={props.genre} /> :
                <ResultsSorter moviesCount={props.movies.length} />}

            {getResultList()}
        </section>
    );
});
