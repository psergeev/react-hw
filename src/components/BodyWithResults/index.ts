import { connect } from 'react-redux';
import { State } from '../../state/reducers';
import BodyWithResults from './BodyWithResults';

const mapStateToProps = (state: State) => ({
    isEmpty: state.isEmpty,
});

export default connect(mapStateToProps)(BodyWithResults);
