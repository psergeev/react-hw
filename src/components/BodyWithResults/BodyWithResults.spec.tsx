import * as React from 'react';
import * as renderer from 'react-test-renderer';
import movies from './__fixtures__/movies';
import BodyWithResults from './BodyWithResults';

jest.mock('../ResultsSorter', () => 'ResultsSorter');
jest.mock('../ResultsDescription', () => 'ResultsDescription');
jest.mock('../MovieCard', () => 'MovieCard');

test('BodyWithResults component works as expected', () => {
    const component = renderer.create(
        <BodyWithResults
            handleMovieCardClick={jest.fn()}
            genre="Fiction"
            movies={movies}
            isEmpty
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('BodyWithResults component shows ResultsSorter without genre', () => {
    const component = renderer.create(
        <BodyWithResults
            handleMovieCardClick={jest.fn()}
            genre=""
            movies={movies}
            isEmpty
        />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
