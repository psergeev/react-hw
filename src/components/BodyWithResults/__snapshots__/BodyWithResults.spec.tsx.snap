// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyWithResults component shows ResultsSorter without genre 1`] = `
<section
  className="results"
>
  <ResultsSorter
    moviesCount={2}
  />
  
</section>
`;

exports[`BodyWithResults component works as expected 1`] = `
<section
  className="results"
>
  <ResultsDescription
    genre="Fiction"
  />
  
</section>
`;
